fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java

```diff
@@ -335,6 +369,45 @@ void checkSelectTableAction(ConnectContext context, Map<TableName, Relation> all
         }
     }
 
+    void checkCanSelectFromColumns(ConnectContext context, Map<TableName, Set<String>> allTouchedTableColumns,
+                                   Map<TableName, Relation> allTouchedTables) {
+        for (Map.Entry<TableName, Set<String>> tableColumns : allTouchedTableColumns.entrySet()) {
+            TableName tableName = tableColumns.getKey();
+            Set<String> columns = tableColumns.getValue();
+            Relation relation = allTouchedTables.get(tableName);
+            Table table;
+            if (relation instanceof TableRelation) {
+                table = ((TableRelation) relation).getTable();
+            } else {
+                table = ((ViewRelation) relation).getView();
+            }
+            if (table instanceof SystemTable && ((SystemTable) table).requireOperatePrivilege()) {
+                try {
+                    Authorizer.checkSystemAction(context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            PrivilegeType.OPERATE);
+                } catch (AccessDeniedException e) {
+                    AccessDeniedException.reportAccessDenied(
+                            InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME,
+                            context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            PrivilegeType.OPERATE.name(), ObjectType.SYSTEM.name(), null);
+                }
+            } else {
+                boolean isView = table instanceof View || table.isMaterializedView();
+                try {
+                    Authorizer.checkColumnsAction(context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            tableName, columns, PrivilegeType.SELECT);
+                } catch (AccessDeniedException e) {
+                    AccessDeniedException.reportAccessDenied(
+                            isView ? InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME : tableName.getCatalog(),
+                            context.getCurrentUserIdentity(), context.getCurrentRoleIds(),
+                            PrivilegeType.SELECT.name(), ObjectType.COLUMN.name(),
+                            String.join(".", tableName.getNoClusterString(),
+                                    String.join(",", columns)));
+                }
+            }
+        }
+    }
+
```

Comment:
Is DeleteStmt need column privilege?

Reply:
For targetTable in DeleteStmt, may not exist the notion of deleting data from specific columns, and it seems that `DeleteStmt` does not have column info. So this time I only check column privilege in subquey of DeleteStmt